Client/Frontend/Home/RecentlyClosedTabsPanel.swift

```diff
@@ -0,0 +1,134 @@
+        tableViewController.view.snp_makeConstraints { make in
+            make.left.right.bottom.equalTo(self.view)
+        }
+
+        tableViewController.didMoveToParentViewController(self)
+    }
+
+    @objc private func historyBackButtonWasTapped(gestureRecognizer: UITapGestureRecognizer) {
+        self.navigationController?.popViewControllerAnimated(true)
+    }
+}
+
+class RecentlyClosedTabsPanelSiteTableViewController: SiteTableViewController {
+    weak var homePanelDelegate: HomePanelDelegate?
+    var recentlyClosedTabs: [ClosedTab] = []
+    weak var recentlyClosedTabsPanel: RecentlyClosedTabsPanel?
+
+    init() {
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tableView.accessibilityIdentifier = "Recently Closed Tabs List"
+        self.recentlyClosedTabs = profile.recentlyClosedTabs.tabs
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = super.tableView(tableView, cellForRowAtIndexPath: indexPath)
+        if let cell = cell as? TwoLineTableViewCell {
+            cell.setLines(recentlyClosedTabs[indexPath.row].title ?? "", detailText: recentlyClosedTabs[indexPath.row].url.absoluteString)
+            cell.imageView?.sd_setImageWithURL((recentlyClosedTabs[indexPath.row].faviconURL ?? "").asURL) { (img, err, type, url) -> Void in
+                guard img != nil else {
+                    if let url = NSURL(string: self.recentlyClosedTabs[indexPath.row].faviconURL ?? "") {
+                        cell.imageView?.setIcon(nil, withPlaceholder: FaviconFetcher.getDefaultFavicon(url))
+                    }
+                    return
+                }
+            }
+        }
+
+        return cell
+    }
+
+    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        guard let homePanelDelegate = homePanelDelegate,
+              let recentlyClosedTabsPanel = recentlyClosedTabsPanel else {
+            log.warning("No site or no URL when selecting row.")
+            return
+        }
+        let visitType = VisitType.Typed    // Means History, too.
+        homePanelDelegate.homePanel(recentlyClosedTabsPanel, didSelectURL: recentlyClosedTabs[indexPath.row].url, visitType: visitType)
+    }
+
+    override func tableView(tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return 0
+    }
+
+    // Functions that deal with showing header rows.
+    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return 1
+    }
+
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // We only display a max of 5 items in this panel
+        return min(5, (profile.recentlyClosedTabs.tabs.count ?? 0))
```

Comment:
And going with the other comment, remove the `min` here and just return the full set of tabs we have in the store. We should enforce the limit at the store layer, not in the view.